### PR TITLE
[Javascript] add sifrr

### PR DESCRIPTION
**Pull request by aadityataparia**
closes https://github.com/the-benchmarker/web-frameworks/issues/1587

Before submitting your PR, please review the following checklist :

## If you are adding a framework

+ [x] A `Dockerfile` exists, or can be built ?
+ [x] I've updated `CI` config file (and **commited** `.travis.yml`)
~~~sh
bin/make ci_config
~~~
+ [x] I've updated `.dependabot` config file  (and **commited** `.dependabot/config.yml`)
~~~sh
bin/make deps_config
~~~
+ [x] All tests passes ?
~~~
export FRAMEWORK=<MY_FRAMEWORK>
shards install
shards build --release
bin/make config
bin/neph ${FRAMEWORK}
crystal spec
~~~

### CODE REVIEW

**Comment by waghanza** [comment]
:tada: awesome

I prefer that folder is named `sifrr` not `sifrr-server`

**Comment by aadityataparia** [comment]
@waghanza hostname is blank, could you please help?

**Comment by waghanza** [comment]
@aadityataparia seems that `node` **12** is no compatible with `uWebSocket`
https://github.com/uNetworking/uWebSockets.js/issues/133

**Comment by waghanza** [comment]
working for `10` and `11`

File: node/sifrr-server/config.yaml [hunk 1]
```diff
@@ -0,0 +1,3 @@
+framework:
+  website: sifrr.github.io/sifrr/#/packages/server/sifrr-server/
```
Comment (waghanza) [inline node/sifrr-server/config.yaml]: seems to be the wrong URL, **SHOULD** be `website: sifrr.github.io/sifrr/#/./packages/server/sifrr-server/`

File: node/sifrr-server/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+WORKDIR /usr/src/app
+
+COPY app.js package.json ./
+
+RUN npm install
+
+ENV NODE_CLUSTER_SCHED_POLICY=rr
+ENV NODE_ENV production
+
+CMD pm2-runtime start app.js -i $(nproc)
```
Comment (waghanza) [inline node/sifrr-server/Dockerfile]: `uWebSockets.js` is not compatible with `node` **12**

Comment (waghanza) [reply node/sifrr-server/Dockerfile]: however @jkyberneees run if on `node` **12** (with a wrapper)  in https://github.com/the-benchmarker/web-frameworks/pull/1570

Comment (jkyberneees) [reply node/sifrr-server/Dockerfile]: Hi @waghanza, I think I have to verify that, here I run it with node v10 https://github.com/TechEmpower/FrameworkBenchmarks/blob/master/frameworks/JavaScript/0http/0http.dockerfile, have you run a successful iteration with node 12?

Comment (jkyberneees) [reply node/sifrr-server/Dockerfile]: Well, I can confirm uWebSockets.js v15.11 works on my Mac using Node v12.7!

Comment (waghanza) [reply node/sifrr-server/Dockerfile]: strange, the binary seems to exists for both `darwin` and `linux` => https://github.com/uNetworking/uWebSockets.js/tree/v15.11.0
but made on **travis** for `osx` and **appveyor** for `linux`

Comment (aadityataparia) [reply node/sifrr-server/Dockerfile]: sifrr with updated uwebsockets is not yet released, I will release and try to update here as well.

File: node/sifrr-server/config.yaml (hunk 1 above)
Comment (aadityataparia) [reply node/sifrr-server/config.yaml]: works for me 🤔

File: node/sifrr-server/Dockerfile (hunk 2 above)
Comment (waghanza) [reply node/sifrr-server/Dockerfile]: I have know 
```javascript
Error: Cannot find module 'graphql'
```

Comment (waghanza) [reply node/sifrr-server/Dockerfile]: when trying to launch on container

**Comment by dalisoft** [comment]
@waghanza uWebSocket is compatible with Node v12. My nanoexpress also uses uWebSocket to power backend

**Comment by waghanza** [comment]
@dalisoft In deed, the version `0.05` of `sifrr`  does not support **node** `12`, but it's now ok with the last version

**Comment by dalisoft** [comment]
Aw, i did not know that, i always update to latest version. But Siffr great library, i am also use this library as dependency to my library

File: node/sifrr-server/Dockerfile (hunk 2 above)
Comment (aadityataparia) [reply node/sifrr-server/Dockerfile]: getting connection timeout now

Comment (aadityataparia) [reply node/sifrr-server/Dockerfile]: works on travis though

File: tools/src/benchmarker.cr [hunk 3]
```diff
@@ -134,7 +134,7 @@ end
 def benchmark(host, threads, connections, duration, target, store) : Filter
   latency = 0.0
   requests = 0.0
-  raw = `#{CLIENT} --url http://#{host}:3000 --init`
+  raw = `#{CLIENT} --url http://localhost:3000 --init`
```
Comment (waghanza) [inline tools/src/benchmarker.cr]: why ?

File: node/sifrr/package.json [hunk 4]
```diff
@@ -0,0 +1,7 @@
+{
+  "dependencies": {
+    "@sifrr/server": "0.0.6",
+    "graphql": "^14.4.2",
```
Comment (waghanza) [inline node/sifrr/package.json]: can you remove this constraint ?

File: node/sifrr/config.yaml [hunk 5]
```diff
@@ -0,0 +1,3 @@
+framework:
+  website: https://sifrr.github.io/sifrr/#/./packages/server/sifrr-server/
+  version: 0.05
```
Comment (waghanza) [inline node/sifrr/config.yaml]: `0.0` instead of `0.05`

File: node/sifrr/package.json (hunk 4 above)
Comment (aadityataparia) [reply node/sifrr/package.json]: version or dependency?

Comment (waghanza) [reply node/sifrr/package.json]: yeh, even use `*` for `graphql` or use a strict version contraint`

even better made this optional when using `sifrr` :stuck_out_tongue:

Comment (aadityataparia) [reply node/sifrr/package.json]: I will, 😂 in next version

File: node/sifrr/config.yaml [hunk 6]
```diff
@@ -0,0 +1,3 @@
+framework:
+  website: https://sifrr.github.io/sifrr/#/./packages/server/sifrr-server/
```
Comment (waghanza) [inline node/sifrr/config.yaml]: remove `https`

File: node/sifrr/package.json (hunk 4 above)
Comment (waghanza) [reply node/sifrr/package.json]: then, I'll please pleased to review a `PR`, reving this dependency :heart:

File: .gitignore [hunk 7]
```diff
@@ -24,3 +24,5 @@ vendor/
 
 # Temporary files
 FRAMEWORKS.yaml
+
```
Comment (waghanza) [inline .gitignore]: not sure it has to be here, but in `node` directory

Comment (aadityataparia) [reply .gitignore]: never liked multiple `.gitignore`, but upto you 👍

**Comment by waghanza** [comment]
@aadityataparia `siffr` has been added to this list (and results are available on `README`)

**Pull request merged**